### PR TITLE
executor, server: load_data.go is changed and add unit test

### PR DESCRIPTION
**Pull request by zoujia-cm**
### What problem does this PR solve?

Problem Summary:
I use "load data" statement to load data from csv file. In csv file, fields are enclosed by "'" and are terminated by ",". lines are terminated by "|"(for example: '11','22'|'33','44'). I generated a csv file with 50000 items of record. When load it into db by "load data" statement, only items less than 50000 (often loss at least one item) were loaded.
### What is changed and how it works?

What's Changed:
The way to locate the terminator (include field terminator and line terminator) was changed.
How it Works:
Line terminator can be located precisely in different situations.
### Related changes

- Need to cherry-pick to the release branch

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- fix load data bug that the number of items loaded by "load data" is less than the actual number

### CODE REVIEW

**Comment by hi-rustin** [comment]
/uncc

Sorry, I am not a reviewer of these SIGs, I suggest you join the [sig-sql-infra slack](https://tidbcommunity.slack.com/archives/CMAKWBNJU), and then join the SIG channel to request a review.

**Comment by zoujia-cm** [comment]
@zimulala  please take a look

File: server/server_test.go [hunk 1]
```diff
@@ -575,6 +575,63 @@ func (cli *testServerClient) runTestLoadDataAutoRandom(c *C) {
 	})
 }
 
+func (cli *testServerClient) runTestLoadDataAutoRandomWithSpecialTerm(c *C) {
+	path := "/tmp/load_data_txn_error_term.csv"
```
Comment (zz-jason) [inline server/server_test.go:579]: I think it's better to convert this test to an integration test. BTW, the `/tmp/load_data_txn_error_term.csv` file is not exists on the repo

**Review by zz-jason — COMMENTED** [review]
how about using` executor, server: support " as field encloser for load data stmt` as the PR title?

**Comment by tiancaiamao** [comment]
The original PR is this one https://github.com/pingcap/tidb/pull/22834

**Comment by tiancaiamao** [comment]
/LGTM
@zimulala

**Pull request merged**